tasks/terraform/src/main/java/com/walmartlabs/concord/plugins/terraform/TerraformTask.java

```diff
@@ -123,16 +125,28 @@ public void execute(Context ctx) throws Exception {
 
     private Backend getBackend(Map<String, Object> cfg) {
         boolean debug = MapUtils.get(cfg, Constants.DEBUG_KEY, false, Boolean.class);
-        String s = MapUtils.getString(cfg, Constants.BACKEND_KEY, DEFAULT_BACKEND);
-        switch (s) {
+        String backendId = DEFAULT_BACKEND;
+        if (cfg.get(Constants.BACKEND_KEY) != null) {
+            if (Map.class.isAssignableFrom(cfg.get(Constants.BACKEND_KEY).getClass())) {
```

Comment:
Perhaps `ContextUtils.assertMap` will be better? Just fail the task call if someone tries to pass some junk as `backend` value?

Reply:
In the case where it's not a Map like "concord" or "none" won't it fail?

Reply:
Ah, I see what you're trying to do. So the `backend` accepts either a String or a Map.